File: base/src/main/java/me/shetj/base/base/BaseModel.java
package me.shetj.base.base;

import android.support.annotation.Keep;

import com.zhouyou.http.model.HttpParams;

import java.util.Map;

import io.reactivex.annotations.NonNull;
import me.shetj.base.http.easyhttp.EasyHttpUtils;

/**
 * <b>@packageName：</b> me.shetj.base.base<br>
 * <b>@author：</b> shetj<br>
 * <b>@createTime：</b> 2018/2/28<br>
 * <b>@company：</b><br>
 * <b>@email：</b> 375105540@qq.com<br>
 * <b>@describe</b><br>
 */

@Keep
public abstract class BaseModel implements IModel {

	@Override
	@NonNull
	public BaseMessage getMessage(int code ,Object obj) {
		BaseMessage message = new BaseMessage();
		message.obj = obj;
		message.type = code;
		return message;
	}

	@NonNull
	public HttpParams getParamsFromMap(@NonNull Map<String,String > map ){
		return EasyHttpUtils.getParamsFromMap(map);
	}

	@Override
	public void onDestroy() {

	}
}

File: base/src/main/java/me/shetj/base/base/BaseModel.kt
package me.shetj.base.base

import android.os.Message
import androidx.annotation.Keep


import io.reactivex.annotations.NonNull
import me.shetj.base.tools.app.getMessage

/**
 * **@packageName：** me.shetj.base.base<br></br>
 * **@author：** shetj<br></br>
 * **@createTime：** 2018/2/28<br></br>
 * **@company：**<br></br>
 * **@email：** 375105540@qq.com<br></br>
 * **@describe**<br></br>
 */

@Keep
abstract class BaseModel : IModel {

    @NonNull
    override fun getMessage(code: Int, obj: Any): Message {
        return Message.obtain().getMessage(code,obj)
    }

    override fun onDestroy() {

    }
}
